### DISCUSSION TITLE
Unpublish programmatic workflows

### DISCUSSION THREAD

**Discussion by truthz03** [elsa-workflows/elsa-core#3899] [Q&A]
Hi,

is it possible to unpublish/publish workflows created with C#?

Our problem is that we want to predefine some workflows inside our code but don't want publish them all.
The reason for this predefined workflows is to get an easy way to publish some of them without the need to create it on your own.

Normally if I create a workflow inside ui designer I have to click publish to enable the workflow und unpublish to disable the workflow again.
Workflows defined in C# are not listed inside the workflow definitions page.
The only way to see these workflows is to open the workflow registry page where I have to select "Programmatic workflow provider"
But on this page it is not possible to unpublish a workflow.

Any ideas?

**Comment by ahossain** [comment]
Looking for a solution to the same issue.

**Reply by truthz03** [reply]
In Elsa v2 this is not possible I think but I found a discussion for Elsa v3
![image](https://user-images.githubusercontent.com/14071920/234191384-6cc6e3ef-e518-4090-a2c7-929c844c2924.png)

I have not tested it now but I hope this is true ^^

**Reply by sfmskywalker** [reply]
Should work, although I haven't tested it. Let me know if it works or not. If not, we'll spend some time to make it work.

**Reply by truthz03** [reply]
If I set the `Published` flag in the SqliteDb the workflow will not be started anymore.
But there is no possibility to unpublish/republish it inside the web designer now or are I'm wrong?

Another problem with code workflows is the "Workflow Instances" page.
The code workflow will be shown and can also be clicked but the informations there are really minimalistic.
![image](https://user-images.githubusercontent.com/14071920/235863914-647af1ff-db11-422e-9e76-61bb409582ee.png)

**Reply by truthz03** [reply]
There seems to be also a problem with code workflows, a cron trigger and the INotificationHandler<WorkflowTriggersIndexed>
If I start the app the first time without an existing SQLiteDB the following things happens with this code workflow
![image](https://user-images.githubusercontent.com/14071920/235888938-f197d398-7398-42ee-9044-832d1daea6a2.png)


1. The ScheduleWorkflows.HandleAsync gets called
![image](https://user-images.githubusercontent.com/14071920/235888381-68cee726-3bf2-4d49-afb2-75da123697db.png)

2. This adds a scheduler
![image](https://user-images.githubusercontent.com/14071920/235888475-e33ed07e-fbd4-4da9-a827-bb2018403fe4.png)

3. After that the CreateSchedulesHostedService.ExecuteAsync gets called
![image](https://user-images.githubusercontent.com/14071920/235888635-2751a1aa-e419-468e-8824-fe65418b0f9c.png)

4. Which also tries to add a scheduler
![image](https://user-images.githubusercontent.com/14071920/235888732-f586a264-1325-4746-8c0e-55e452f70bd1.png)

5. But now you get an error because there is already a job with the same identity and the app stops


If I start the app again with a existing SQLiteDb now point 1 & 2 will not happen and everything is

**Reply by sfmskywalker** [reply]
@truthz03 Any chance you can break down your findings into separate issues, with steps to reproduce (and ideally a sample application and/or workflows)?

**Reply by truthz03** [reply]
@sfmskywalker I have created 2 issus for this:
https://github.com/elsa-workflows/elsa-core/issues/4034
https://github.com/elsa-workflows/elsa-core/issues/4035
